contracts/abstracted_account.algo.ts

```diff
@@ -42,6 +42,20 @@ export class AbstractedAccount extends Contract {
    * Verify the contract account is not rekeyed
    */
   verifyAppAuthAddr(): void {
+    /** dont force the caller to rekey in a separate txn
+     * if this gets called and its not rekeyed back
+     * then we'll do it in an itxn
+     */
+    if (this.app.address.authAddr !== globals.zeroAddress) {
+      sendPayment({
+        sender: this.app.address.authAddr,
```

Comment:
I don't think this is doing what you think it is. Because `sender` is `this.app.address.authAddr`, this itxn is rekeying the *plugin contract address*, rather than the abstracted account. 

You can see this if you delete the rekey in a plugin (ie. delete this line https://github.com/kylebeee/account_abstraction/blob/7ab2011b03a6400621ad44563a261418537315b0/contracts/optin_plugin.algo.ts#L20) and run the tests. You will get `transaction 4RFR4HUTBACPUQJUC47ECG7VTF4ORFUVBGVVBIAJTGDUETUAVRPQ: logic eval error: app 21352 (addr EKQW4HIKON7I3MKXYCIQANYLEIU3VPPDGTCUZTGXFM2JYQRN6G5X644Q7Q) unauthorized DO6THNPZQ7ISTVEIKOPDG3ICJLOVJXGZJ37TYVKPVAI4YBUQS42ZTXZSKA`

If `this.app.address.authAddr !== globals.zeroAddress` then this app no longer has control of its app account, thus it cannot undo a rekey.

Reply:
hmmm, it would literally have to be a standalone txn to rekey back specifically wouldn't it

Reply:
Yeah it has to be rekeyed in a transaction "signed" by whoever controls the auth addr at the end of the group (whether that's some EOA or an app account). By asserting `this.app.address.authAddr === globals.zeroAddress`, rather than checking for an explicit rekey, we're allowing this rekey to happen anywhere. It can be an itxn at an arbitrary depth or it can be from EOA in the outer txn. The only thing that matters is that the app has control of its own account, but it doesn't matter how that happens.

The alternative approach is to assert the last txn in the group is a rekey back to `this.app.address`. Doing this though limits the rekey to being in the outer group, rather than being done by an app.

Reply:
Im trying to think of a way we can get the best of both worlds. There will certainly be some edge cases where this wont work with pre-existing apps because it has this limitation of needing to rekey back and forth. What I'd like to do is have us 
 - be able to let the external caller rekey during any part of the group
 - allow the rekey to happen on that last txn in the group so we have 14 open slots instead of 13

With the initial approach i think it meant any groups need to be aware of the rekey responsibility and itll make composability a bit harder.

Reply:
Ah yeah that makes sense. Why waste a txn in the outer when it could be the rekey itself. Check out the commit I just pushed to this PR